apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerDataReader.cs

```diff
@@ -323,13 +318,35 @@ private SpannerDbType GetSpannerFieldType(int i)
             return SpannerDbType.FromProtobufType(fieldMetadata.Type);
         }
 
+        /// <summary>
+        /// Common code used by both Dispose and Close. Dispose can't call Close, as it's not
+        /// present in .NET Standard 1.5; Close can't call dispose as we don't want to call into
+        /// the base Dispose method.
+        /// </summary>
+        private void CloseImpl()
+        {
+            lock (_lock)
+            {
+                if (_closed)
+                {
+                    return;
+                }
+                _closed = true;
+                Logger.LogPerformanceCounter(
+                    "SpannerDataReader.ActiveCount",
+                    () => Interlocked.Decrement(ref s_readerCount));
+
+                _resultSet?.Dispose();
+                _resourceToClose?.Dispose();
+            }
+        }
+
 #if !NETSTANDARD1_5
 
         /// <inheritdoc />
         public override void Close()
         {
-            _resultSet?.Close();
-            _connectionToClose?.Close();
+            CloseImpl();
```

Comment:
This wasn't disposing `_txToClose`, but now is. Is this intentional?

Reply:
Yes - that was a bug before, and this PR is hoping to avoid that happening again :)

Reply:
Oh, haha...guess I should have actually read the PR title.
LGTM